### PR TITLE
Introduce ElementExtractorTransformer

### PR DESCRIPTION
**Pull request by hugo-goncalves-kununu**
- Introduce ElementExtractorTransformer

- Add unit tests of ElementExtractorTransformer

- Add more unit tests coverage to other transformers

### CODE REVIEW

File: src/Transformer/ElementExtractorTransformer.php [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+                // We already have found a <splitTag> so we are storing all the next elements
+                // after the <splitTag> (until the next one only!)
+                $elements[$id][self::FIELD_NEXT][] = $child;
+            }
+        }
+
+        return $elements;
+    }
+
+    private function findInnerElement(AbstractNode $node): ?AbstractNode
```
Comment (kununu-drocha) [inline src/Transformer/ElementExtractorTransformer.php:97]: how about findSplitTagElement for this method. I think it better matches the descrition on top of file

**Review by kununu-drocha — COMMENTED** [review]
looks great. Just some questions and suggestions

File: tests/Transformer/ElementExtractorTransformerTest.php [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+HTML
+                ,
+            ],
+            'inner_image_inside_p'                        => [
+                <<<'HTML'
+<p>Here is an <b><i><a href="www.kununu-com">inner <img src="kununu.gif"> <b> image</b></a></i></b> that should come after the a</p>
+HTML
+                ,
+                <<<'HTML'
+<p>Here is an </p><p><b><i><a href="www.kununu-com">inner <img src="kununu.gif"><b> image</b></a></i></b></p><p> that should come after the a</p>
```
[the remark this answers is not in the document]
Comment (hugo-goncalves-kununu) [reply tests/Transformer/ElementExtractorTransformerTest.php:97]: In those cases yes.

File: src/Transformer/ElementExtractorTransformer.php [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+ * This content transformer will extract an element inside another element.
+ * Basically it splits a <searchTag> into multiple <searchTag> tags separated by the <splitTag> tag).
+ *
+ * Example:
+ * <searchTag>A text node and <splitTag>My Split Tag</splitTag> and another text node</searchTag>
+ *
+ * Is transformed to:
+ * <searchTag>A text node and </searchTag><splitTag>My Split Tag</splitTag><searchTag> and another text node</searchTag>
+ *
+ * It also allow to search for the <searchTag> that are stored in inner tags by passing an array of allowed inner tags in the constructor.
```
[the remark this answers is not in the document]
Comment (hugo-goncalves-kununu) [reply src/Transformer/ElementExtractorTransformer.php:18]: You're right!

**Review by kununu-drocha — APPROVED** [review]
:top: :top: :top:

**Pull request merged**